src/bokeh/models/annotations/legends.py

```diff
@@ -531,6 +538,28 @@ def js_on_click(self, handler: JsEventCallback) -> None:
         """ Set up a JavaScript handler for legend item clicks. """
         self.js_on_event(LegendItemClick, handler)
 
+class Val(HasProps):
+    """ Represents a single data point. """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, value: Init[float | str] = Intrinsic) -> None: # TODO str -> FactorType
+        super().__init__(value=value)
+
+    # TODO frozen property; allow to be set once
+    value = Required(CoordinateLike, help="""
+    The value.
+    """)
+
+X = Y = Either(Float, Instance(Val)) # screen/absolute or data
```

Comment:
No, because they are either used in immutable (as a part of other types) or cloned contexts (at the top-level; they will be cloned with `__call__` if they need to be modified, e.g. `help` or `default` added).

Reply:
The class values are more like recipes that the `HasProps` metaclass uses to add the necessary descriptors to model instances upon their creation, so I agree that this should work fine. That said, it does look a bit weird, since AFAIK no other models do this. So my slight preference would be to *not* write things this way, in order to be consistent with usage everywhere else